routers/api/v1/activitypub/person.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package activitypub
+
+import (
+	"net/http"
+	"strings"
+
+	"code.gitea.io/gitea/modules/activitypub"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/routers/api/v1/user"
+
+	ap "github.com/go-ap/activitypub"
+)
+
+// Person function
+func Person(ctx *context.APIContext) {
+	// swagger:operation GET /activitypub/user/{username} activitypub activitypubPerson
+	// ---
+	// summary: Returns the person
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: username
+	//   in: path
+	//   description: username of the user
+	//   type: string
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/ActivityPub"
+
+	user := user.GetUserByParamsName(ctx, "username")
+	if user == nil {
+		return
+	}
+	username := ctx.Params("username")
+
+	link := strings.TrimSuffix(setting.AppURL, "/") + strings.TrimSuffix(ctx.Req.URL.EscapedPath(), "/")
+	person := ap.PersonNew(ap.IRI(link))
+
+	name := ap.NaturalLanguageValuesNew()
+	err := name.Set("en", ap.Content(username))
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "Set name", err)
+		return
+	}
+	person.Name = name
+
+	person.Inbox = ap.Item(ap.IRI(link + "/inbox"))
```

Comment:
And again, there is a const value for [`ap.Inbox`](https://pkg.go.dev/github.com/go-ap/activitypub#Inbox). So you can just write `ap.Inbox.AddTo(person)`

Reply:
nice to see the maintainer of the lib reviewing this pull - thanks :heart:

Reply:
@mariusor Did I do this correctly in 528c282d686f9ced4b6e26b73cdf2c0e5b97bfce? I had to first set the inbox and outbox to `Nil` for `AddTo()` to work.

Reply:
You don't need to nil the property first I think.

Reply:
I think `PersonNew()` sets the inbox and outbox properties to `{"id":"test-inbox","totalItems":0,"type":"OrderedCollection"}` or something like that by default, which is why I had to change them to `Nil`.

Reply:
Yes. I remember now. :) 

I think the behaviour on the `XXXNew()` initialization functions is somewhat obsolete (as in the inbox, outbox are created as `ap.Collection` objects instead of IRIs or being unset), thank you for bringing this to my attention. I will probably change them at one point. 

I realized in my own code I just instantiate the objects inline:

```go
person := ap.Person{
    ID: ap.IRI("htttp://example.com/"),
    Name: name,
    Inbox: inboxIRI, // etc
}
```

Reply:
Another thing to keep in mind (and it's one of the reasons why I didn't advertise the libraries too much) is the fact that I haven't really released a "stable" version of them. So the APIs might change in some ways. Just from our conversation here, I found two issues to fix.

I would advise that you create some wrapper functionality for all of your interactions with go-ap packages and when I make changes there will be a smaller number of places to update. 

Examples of API changes would be the `XXXNew()` functions would have slightly different initialization mechanism, and possibly be replaced with a fluent API of building the object, something like:

```go
person := ap.Builder().WithID(id).OfType(ap.PersonType).WithNameMap(names).WithInbox(inboxIRI).Build()

// OR
// ap.NewPerson(type ap.ActivityVocabularyType, propSetterFn ...setterFn) *ap.Person

person := ap.NewPerson(type, WithID(iri), WithNameMap(names), WithDefaultInbox())
```
Feel free to experiment and let me know which one you would find clearer.

Reply:
@mariusor 

a good way to maintain a beta lib while breaking thigs would be, to tag at least before a breaking change gets done 
and tag a v0.?.? so it's clear that it's beta

optional plus points would be to either create a pull with version bumg and ajustment to new api to bigger projects afterwards, or write a migration guide in some doc (example: https://gitea.com/gitea/go-sdk/src/branch/master/docs/migrate-v0.14-to-v0.15.md)

Reply:
@6543 thank you, but all of that is a lot of friction that I'm not willing to go through for every change I might want to implement from here until I'm satisfied with the API of the libraries. Therefore the warning.